Item durability loss now depends on the item used.

bearbin opened the pull request:
Armour durability also no longer changes when it is used to break blocks or attack mobs.

Fixes #4119

bearbin commented on src/Items/ItemHandler.h:
```diff
@@ -21,10 +21,24 @@ class cItemHandler
 public:
 
-	enum eDurabilityLostAction
+	enum eDurabilityLostAction : int
```
That would probably make sense, I just included it here as that's where it was before.

bearbin commented:
> Overall I'm not sure it makes sense to add so many different categories of item use, especially when most only make sense for one item. Shouldn't all knowledge about particular items be in their own handler?

I agree the system as it is is rather convoluted and there are too many categories of item use, but I couldn't really think of a much better solution. If you have any ideas I'd be glad to hear them.

bearbin commented on src/Items/ItemElytra.h:
```diff
@@ -0,0 +1,40 @@
+
+// ItemElytra.h
+
+// Declares the various flint and steel ItemHandlers
```
In fact, this comment is completle useless, I'm going to get rid of it on the other item handlers as well (most have no comment at all).

peterbell10 commented:
Some of the actions only happen when an item is used normally:  `dlaCarrotOnStickBoost`, `dlaElytraFly`, `dlaShootArrow`, `dlaTilling` and all the fishing ones.  I would think these could be handled directly within the item handlers' `OnItemUse` functions.

`dlaShearBlock` is only used for shears where it's the exact same damage as the other break block actions.  I would just not add it at all.

`dlaLightFire` and `dlaShearSheep` are both handling using an item on an entity.  Maybe give them a more generic name like `dlaUseEntity`, in which case they can share the same action.

This leaves `dlaShieldBlock` which doesn't really fit into `GetDurabilityLossByAction` because as the wiki says:
>When the shield blocks an attack, it takes damage equal to the strength of the attack (rounded down) plus 1.

bearbin commented:
I believe that I have now addressed your comments. I removed all the damage categories except where the damage taken for an action could depend on the item used.

peterbell10 commented on src/Entities/Player.h:
```diff
@@ -413,7 +414,16 @@ class cPlayer :
+	void UseEquippedItem(short a_Damage);
+
+	/** Shorthand for UseEquippedItem(1) */
+	void UseEquippedItem(void);
```
What was wrong with using a default value?

bearbin replied:
That's a much better idea, I didn't think about the possibility.
